### PR TITLE
Fix vulnerabilities by upgrading dependencies 

### PR DESCRIPTION
**Pull request by CatherineF-dev**
/hold

### CODE REVIEW

**Comment by CatherineF-dev** [comment]
Have some unit test failures, will update test codes.

File: custom-metrics-stackdriver-adapter/adapter.go [hunk 1]
```diff
@@ -144,12 +146,16 @@ func (sa *StackdriverAdapter) withCoreMetrics(translator *translator.Translator)
...
-	nodes := informers.Core().V1().Nodes()
-	if err := api.Install(provider, pods.Lister(), nodes.Lister(), server.GenericAPIServer); err != nil {
+	podInformer, nil := informers.ForResource(corev1.SchemeGroupVersion.WithResource("pods"))
+	if err != nil {
 		return err
 	}
 
+	nodes := informers.Core().V1().Nodes()
+	if err := api.Install(provider, podInformer.Lister(), nodes.Lister(), server.GenericAPIServer, []labels.Requirement{}); err != nil {
+		return nil
```
Comment (leonzz) [inline custom-metrics-stackdriver-adapter/adapter.go:156]: you're returning nil error in a non-nil error check, likely a typo?

File: custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/provider.go [hunk 2]
```diff
@@ -49,9 +50,83 @@ type coreClientInterface interface {
 	getNodeRAM(nodesNames []string) (map[string]resource.Quantity, map[string]api.TimeInfo, error)
 }
 
+func (p *CoreProvider) GetPodMetrics(pods ...*metav1.PartialObjectMetadata) ([]metrics.PodMetrics, error) {
```
Comment (leonzz) [inline custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/provider.go:53]: NIT: comment exported methods for readability and go linters. same thing for GetNodeMetrics

File: custom-metrics-stackdriver-adapter/adapter.go (hunk 1 above)
Comment (CatherineF-dev) [reply custom-metrics-stackdriver-adapter/adapter.go]: Done.

File: custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/provider.go (hunk 2 above)
Comment (CatherineF-dev) [reply custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/provider.go:54]: Done.

**Review by leonzz — APPROVED** [review]

**Comment by CatherineF-dev** [comment]
/hold 

Need to wait another dependency fix.

**Comment by CatherineF-dev** [comment]
Tested locally, got error

```
F0425 18:58:49.422639       1 openapi.go:43] Failed to build open api spec for root: cannot find model definition for k8s.io/metrics/pkg/apis/custom_metrics/v1beta1.MetricValueList. If you added a new type, you may need to add +k8s:openapi-gen=true to the package or type and run code-gen again
```

**Comment by CatherineF-dev** [comment]
Tested it can work.

```bash
Annotations:                <none>
Namespace:                  default
Labels:                     <none>
Annotations:                <none>
CreationTimestamp:          Mon, 29 Apr 2024 17:31:33 +0000
Reference:                  Deployment/custom-metric-sd
Metrics:                    ( current / target )
  "custom-metric" on pods:  40 / 20
Min replicas:               1
Max replicas:               5
Deployment pods:            2 current / 2 desired
Conditions:
  Type            Status  Reason              Message
  ----            ------  ------              -------
  AbleToScale     True    ReadyForNewScale    recommended size matches current size
  ScalingActive   True    ValidMetricFound    the HPA was able to successfully calculate a replica count from pods metric custom-metric
  ScalingLimited  False   DesiredWithinRange  the desired count is within the acceptable range
Events:           <none>
```

**Comment by CatherineF-dev** [comment]
Discussed offline, leonzz has LGTMed. Will merge.

**Pull request merged**